var/spack/repos/builtin/packages/boost/package.py

```diff
@@ -254,6 +254,13 @@ def determine_b2_options(self, spec, options):
                 'toolset=%s' % self.determine_toolset(spec)
             ])
 
+        # clang is not officially supported for pre-compiled headers
+        # and at least in clang 3.9 still fails to build
+        #   http://www.boost.org/build/doc/html/bbv2/reference/precompiled_headers.html
+        #   https://svn.boost.org/trac/boost/ticket/12496
+        if spec.satisfies('%clang'):
```

Comment:
my point is, you observe that it's broken for `3.9`, but disable it for all versions. I think some version range should be added either way and adjusted the upper bound later if needed.